### PR TITLE
fix cluster role getter in rbac for multitenancy

### PR DESCRIPTION
**Pull request by pdgetrf**
fix an issue where clusterrolebinding was unable to retrieve cluster role for a tenant.

### CODE REVIEW

File: pkg/registry/rbac/validation/rule.go [hunk 1]
```diff
@@ -408,6 +409,15 @@ func (r *StaticRoles) GetClusterRole(name string) (*rbacv1.ClusterRole, error) {
 	return nil, errors.New("clusterrole not found")
 }
 
+func (r *StaticRoles) GetClusterRoleWithMultiTenancy(tenant, name string) (*rbacv1.ClusterRole, error) {
+	for _, clusterRole := range r.clusterRoles {
+		if clusterRole.Name == name && clusterRole.Tenant == tenant{
+			return clusterRole, nil
+		}
+	}
+	return nil, errors.New("clusterrole not found")
```
Comment (chenqianfzh) [inline pkg/registry/rbac/validation/rule.go:418]: Minor: better to include the clusterrole name and tenant name in the error message.

File: plugin/pkg/auth/authorizer/rbac/subject_locator_test.go [hunk 2]
```diff
@@ -115,7 +124,7 @@ func TestSubjectLocator(t *testing.T) {
 				},
 				{
 					// verb matchies correctly
```
Comment (chenqianfzh) [inline plugin/pkg/auth/authorizer/rbac/subject_locator_test.go:126]: typo: matchies

File: pkg/registry/rbac/validation/rule.go (hunk 1 above)
Comment (pdgetrf) [reply pkg/registry/rbac/validation/rule.go:418]: fixed

File: plugin/pkg/auth/authorizer/rbac/subject_locator_test.go (hunk 2 above)
Comment (pdgetrf) [reply plugin/pkg/auth/authorizer/rbac/subject_locator_test.go:126]: fixed

File: plugin/pkg/auth/authorizer/rbac/rbac_test.go [hunk 3]
```diff
@@ -44,18 +46,18 @@ func newRole(name, namespace string, rules ...rbacv1.PolicyRule) *rbacv1.Role {
 	return &rbacv1.Role{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name}, Rules: rules}
```
Comment (chenqianfzh) [inline plugin/pkg/auth/authorizer/rbac/rbac_test.go:46]: Role is namespace-scoped, so it "Tenant" should be set.

Comment (pdgetrf) [reply plugin/pkg/auth/authorizer/rbac/rbac_test.go:46]: fixed

**Pull request merged**